controller_adapters/database.go

```diff
@@ -82,3 +84,89 @@ func (d *Database) GetProfileResolvers() ndb_api.ProfileResolvers {
 	return profileResolvers
 
 }
+
+// Returns basic details about the Time Machine if provided in the
+// underlying database, else returns defaults like:
+// TM Name: db_instance_name_TM
+// TM Description: Time Machine for db_instance_name
+// SLA Name: NONE
+func (d *Database) GetTMDetails() (tmName, tmDescription, slaName string) {
+	tmInfo := d.Spec.Instance.TMInfo
+
+	tmName = tmInfo.Name
+	tmDescription = tmInfo.Description
+	slaName = tmInfo.SLAName
+
+	if tmName == "" {
+		tmName = d.GetDBInstanceName() + "_TM"
+	}
+	if tmDescription == "" {
+		tmDescription = "Time Machine for " + d.GetDBInstanceName()
+	}
+	if slaName == "" {
+		slaName = common.SLA_NAME_NONE
+	}
+
+	return
+}
+
+// Returns a schedule struct for the time machine.
+func (d *Database) GetTMSchedule() (schedule ndb_api.Schedule, err error) {
```

Comment:
Please revisit the field-mappings as per the offline discussions:
eg Quarterly schedule is not applicable for NONE sla or BRASS sla. 

There are also other multiple fields that UI "greys out" (based on applicability) when you select a specific SLAs from the dropdown. 

The easiest way would be to take a look at the UI code and how it disables those input fields based on the SLA selection and translate that into your logic here.

Reply:
Cool, that is a great catch.. Thanks! :)

Reply:
So I looked into the UI code as suggested. Although the fields are greyed out for certain SLA selections, the API request payload does not change. This means that the API is ignoring the non-mandatory inputs for these SLAs. 

To confirm, I provisioned a database using a REST API call via postman with a NONE SLA and non-empty values for the fields in the schedule. The provisioning went through successfully. 

What should we be doing in this scenario?

Reply:
By greying out the fields based on the SLA selected, the UI makes it explicit to the user that those "fields are not required." How do we similarly make it explicit to the user using the NDB operator. Webhooks is one option here. So those validations may need to be implemented once the webhooks PR is merged. That way, if and when the validation is added in the APIs in the future, we will not need to change the operator code.  

For now, you may choose to send the default values for those keys in the payload since it has been verified that sending default values for those non-applicable does not adversely impact the provisioning flow. Also the same is being done by the UI code too.